File: core/src/ship.rs
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum ShipType {
    Military,
    Spacelift,
}

#[derive(Debug, Clone)]
pub struct Ship {
    pub ship_type: ShipType,
    pub is_crippled: bool,
}

impl Ship {
    pub fn new(ship_type: ShipType, is_crippled: bool) -> Self {
        Ship {
            ship_type,
            is_crippled,
        }
    }

    pub fn can_cross_restricted_lane(&self) -> bool {
        match self.ship_type {
            ShipType::Military => !self.is_crippled,
            ShipType::Spacelift => false,
        }
    }
}

File: src/engine/types/military/ship.nim
## Ship Type Definitions for EC4X
##
## This module contains the type definitions for individual ships, including their
## stats, cargo, and operational capabilities.

import std/[options]
import ../../../common/types/units

export ShipClass, ShipType, ShipStats, ShipRole

type
  CargoType* {.pure.} = enum
    ## Type of cargo loaded on transport ships
    None,
    Marines,      # Marine Division (MD) - TroopTransport
    Colonists,    # Population Transfer Unit (PTU) - ETAC
    Supplies      # Generic cargo (future use)

  ShipCargo* = object
    ## Cargo loaded on transport ships (ETAC/TT)
    cargoType*: CargoType
    quantity*: int          # Number of units loaded (0 = empty)
    capacity*: int          # Maximum capacity (CL = Carry Limit)

  Ship* = object
    ## Ship representation with full combat and operational stats
    ## Used for all ship types: combat, intel, expansion, auxiliary,
    ## fighter
    shipClass*: ShipClass
    shipType*: ShipType      # Military or Spacelift (transport)
    stats*: ShipStats
    isCrippled*: bool
    name*: string            # Optional ship name
    cargo*: Option[ShipCargo]  # Cargo for ETAC/TT (Some), None for
                               # combat ships
